port/zos390/omrvmem.c

```diff
@@ -1352,9 +1363,23 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
 		uintptr_t defaultLargePageSize = 0;
 		uintptr_t defaultLargePageFlags = OMRPORT_VMEM_PAGE_FLAG_NOT_USED;
 
-		/* If the page type is NOT_USED (as with the -Xlp<size> options)
-		 * the legacy behavior is to default to FIXED pages.
-		 */
+		/* If the page type is PREFER_PAGEABLE try to get 1M pageable large pages. Otherwise, use a fixed large page. */
+		if ((IS_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE(validPageFlags))) {
+
+			/* Try 1M large pages */ 
+			if (ONE_M == validPageSize) {
+				pageSizeFound = isLargePageSizeSupported(portLibrary, ONE_M, OMRPORT_VMEM_PAGE_FLAG_PAGEABLE);
```

Comment:
Later on in this function we potentially can do exactly the same call again. Would be better to try to avoid this?

Reply:
We can easily guard the second invocation upon checking the `validFlags` for `PREFER_PAGEABLE`.  Since `validFlags` is reset on the first invocation, it wont be called on the second.

Reply:
This can easily be done by creating a Boolean in the method like `isPageableAttempted` which starts off as LOW.  Then we can turn that bit HIGH when using the first invocation `isLargePageSizeSupported(..pageable)`.

On the second Invocation of `isLargePageSizeSupported(..pageable)`, we guard it if `isPageableAttempted` bit is LOW.